### ISSUE TITLE
Hide `Toggle Fullscreen` for MacOS which adds its own fullscreen menu item

### ISSUE THREAD

**Issue by lucyleeow** [napari/napari#6240]
Our `Toggle Fullscreen` view menubar item works but the check mark does not work. MacOS adds its own full screen item to view menu: https://github.com/napari/napari/issues/5092#issuecomment-1252882551, so we should just hide ours.

Suggested here: https://github.com/napari/napari/pull/6137#issuecomment-1726764083

**Comment by psobolewskiPhD** [comment]
Disabling our Full screen is easy (I have a branch), but there is a small issue:
the shortcut
The menu item has `        keybindings=[StandardKeyBinding.FullScreen],`
which I guess is from app_model?
It shows Control-Command-F, which works when the menu item is enabled.
...but it isn't actually the macOS standard keybinding for Full screen post Monterey (macOS 12), when it was changed to Fn (Globe) F.

So if we disable our Full screen, then only the macOS standard will work, based on OS version. So someone who learned Full Screen napari binding may be confused when it stops working?
Now they could change it in their OS settings...

I think in an ideal world, the old keybind would keep working (and be settable as a Viewer keybind), but there would be no custom menu item, just the native one?

**Comment by psobolewskiPhD** [comment]
What's weird is other apps, such as Safari or Calendar will respect either Ctrl-Command-F or Fn-F. But if I disable the non-native menu item in napari, then Ctrl-Command-F stops working.

**Comment by lucyleeow** [comment]
> Disabling our Full screen is easy (I have a branch)

Please feel free to put up a PR then! Do you add a condition to `'when'` field in 'menu' field in `Action`?

> I think in an ideal world, the old keybind would keep working (and be settable as a Viewer keybind),

I think that since we register the action, even if the menu item is not visible, it would be available in the command palette (when this feature gets added) but I am not sure if it would be 'editable' in the command palette. I would think that since we are only hiding this menu item, the action should still be useable via the command palette. 

Potentially, if system is mac OS post Monterey, we can add a full screen action that does not get added to any menu, I think this would mean that the keybinding would work as long as this action is enabled.

Are you interested in doing this?

> The menu item has         keybindings=[StandardKeyBinding.FullScreen],
which I guess is from app_model?

Yes, for reference `StandardKeyBinding.FullScreen` is from app-model and is defined here:

https://github.com/pyapp-kit/app-model/blob/cd8b33da58ffbc9a7b5713bf5c6ddbed35899ff1/src/app_model/types/_keys/_standard_bindings.py#L107

I think for mac it is: `KeyMod.WinCtrl | KeyMod.CtrlCmd | KeyCode.KeyF`

**Comment by lucyleeow** [comment]
> What's weird is other apps, such as Safari or Calendar will respect either Ctrl-Command-F, even if they show Fn-F. But if I disable the non-native menu item in napari, then Ctrl-Command-F stops working in napari.

Yes that is weird, do we know why mac adds its own full screen menu item? (:roll_eyes: apple)

**Comment by psobolewskiPhD** [comment]
> > What's weird is other apps, such as Safari or Calendar will respect either Ctrl-Command-F, even if they show Fn-F. But if I disable the non-native menu item in napari, then Ctrl-Command-F stops working in napari.
> 
> Yes that is weird, do we know why mac adds its own full screen menu item? (🙄 apple)

Yeah it's weird -- the Window menu for example doesn't have any of the native menu items, but Help does have the Search widget (which is super handy when having many plugins).

**Comment by lucyleeow** [comment]
Interesting, thanks! Hiding the toggle menu item is fine, trying to retain access to the full screen keybinding while hiding the menu item is the tricky part! See  #6327

**Comment by lucyleeow** [comment]
I think it would be best to wait until #6204 is in, so we can add the keybinding via app-model, before we address this.

See: https://github.com/napari/napari/pull/6327#discussion_r1363847884
